src/wallet/wallet2.cpp

```diff
@@ -2332,6 +2333,8 @@ bool wallet2::delete_address_book_row(std::size_t row_id) {
 //----------------------------------------------------------------------------------------------------
 void wallet2::refresh(uint64_t start_height, uint64_t & blocks_fetched, bool& received_money)
 {
+  key_ref kref(*this);
```

Comment:
Is this needed?

Reply:
No, but it speeds up saving rings from incoming txes if there are several.